### PR TITLE
Created InputHandler.js

### PR DESCRIPTION
**Pull request by elijah-t**
The InputHandler is mostly finished, although there may be things missing or not tested.

edit: InputHandler, not InputCreator

### CODE REVIEW

File: InputHandler.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+            y: 0
+        }
+
+        this.keyUp = this.keyUp.bind(this);
+        this.keyDown = this.keyDown.bind(this);
+
+        this.mouseDown = this.mouseDown.bind(this);
+        this.mouseUp = this.mouseUp.bind(this);
+
+        this.x = this.mouseMoveX.bind(this);
```
Comment (Squishy123) [inline InputHandler.js]: Did you mean:
```
this.mouseMoveX = this.mouseMoveX.bind(this)
this.mouseMoveY = this.mouseMoveY.bind(this)
```

I don't see you using this.x or this.y after this.

File: InputHandler.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        this.y = this.mouseMoveY.bind(this);
```
Comment (Squishy123) [inline InputHandler.js]: Did you mean:
``` suggestion
this.mouseMoveX = this.mouseMoveX.bind(this)
this.mouseMoveY = this.mouseMoveY.bind(this)
```

I don't see you using this.x or this.y after this.

File: InputHandler.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        this.input.mouse[event.which] = false;
+    }
+
+    mouseMoveX(event) {
+        this.input.x = event.clientX;
+    }
+
+    mouseMoveY(event) {
+        this.input.y = event.clientY;
+    }
```
Comment (Squishy123) [inline InputHandler.js]: Let's combine the mousemove callback into one, so it sets input.x and input.y under one function
```suggestion
mouseMove(event) {
this.input.x = event.clientX;
this.input.y = event.clientY;
}
```

File: InputHandler.js [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+    startHandler() {
+        this.elem.addEventListener("keydown", this.keyDown);
+        this.elem.addEventListener("keyup", this.keyUp);
+        this.elem.addEventListener("mousedown", this.mouseDown);
+        this.elem.addEventListener("mouseup", this.mouseUp);
+        this.elem.addEventListener("mousemove", this.mouseMoveX);
+        this.elem.addEventListener("mousemove", this.mouseMoveY);
```
Comment (Squishy123) [inline InputHandler.js]: Combine the callback to one eventlistener here:
```suggestion
this.elem.addEventListener("mousemove", this.mouseMove);
```

File: InputHandler.js [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+        this.input.keys[event.which] = false;
+    }
+
+    mouseDown(event) {
+        this.input.mouse[event.which] = true;
+    }
+
+    mouseUp(event) {
+        this.input.mouse[event.which] = false;
+    }
```
Comment (Squishy123) [inline InputHandler.js]: Change to alternative:
event.which is deprecated
see [API](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/which)

File: InputHandler.js [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+
+    keys() {
+        return this.input.keys;
+    }
+
+    mouse() {
+        return this.input.mouse;
+    }
+
+    xpos() {
```
Comment (Squishy123) [inline InputHandler.js]: Small name change
```suggestion
mouseX() {
```

File: InputHandler.js [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+        return this.input.x;
+    }
+
+    ypos() {
```
Comment (Squishy123) [inline InputHandler.js]: Small name change
```suggestion
mouseY() {
```

**Review by Squishy123 — CHANGES_REQUESTED** [review]
Found a few things to change. I'll do a more in-depth review later :) Good Work so far!

**Comment by Squishy123** [comment]
Added a few changes to the class:

1. We can actually avoid having to manually bind the "this" scope to each class function by using arrow functions:

e.g.
 ```
classFunct = (param) => {
...
}
```

2. You can export the classes in 1 line:
```
export default class ClassName {
...
}
```

Also a general heads-up, 
- try to keep filenames all lowercase
- variables and functions camelcase
- classes pascal case

**Comment by elijah-t** [comment]
Alright cool, I'll keep that in mind as well 👍

**Comment by Squishy123** [comment]
Gonna get you to write some tests for practice. Just create a demo in the /tests/inputHandlerTest.js to test keyboard and mouse input. You can write stuff to the console or display elements on the page if you want.

**Pull request closed without merging**